cylc/flow/cycling/iso8601.py

```diff
@@ -647,111 +666,147 @@ def ingest_time(value, my_now=None):
         # Expanded year
         return value
 
+    value = value.replace(" ", "")  # remove extraneous whitespace
     parser = SuiteSpecifics.point_parser
     offset = None
 
-    if my_now is None:
-        my_now = parser.parse(get_current_time_string())
+    # work out what transformations we need to apply
+    is_prev_next = "next" in value or "previous" in value
+    is_offset = value.startswith("P") or value.startswith("-P")
+    is_truncated = False
+    if not (is_prev_next or is_offset):
+        timepoint = parser.parse(value)
+        is_truncated = timepoint.truncated
+
+    if is_prev_next or is_offset or is_truncated:
+        # we need information about the current time to process this
+        # this should only be attempted if the calendar is in
+        # gregorian mode
+        if CALENDAR.mode != Calendar.MODE_GREGORIAN:
+            raise CylcConfigError(
+                f'The value "{value}" can not be calculated '
+                f'in {CALENDAR.mode}'
+            )
+
+        if now is None:
+            now = parser.parse(get_current_time_string())
+        else:
+            now = parser.parse(now)
+
+    if is_prev_next:
+        # e.g: prev(-01)
+        cycle_point, offset = _prev_next(value, now, parser)
+    elif is_offset:
+        # e.g: +P1D
+        cycle_point = now
+        offset = value
+    elif is_truncated:
+        # e.g: 00T00
+        cycle_point = now + timepoint
     else:
-        my_now = parser.parse(my_now)
+        # e.g. 2000-01-01T00:00Z
+        return value
+
+    if offset is not None:
+        # add/subtract offset duration to/from chosen timepoint
+        duration_parser = SuiteSpecifics.interval_parser
+
+        offset = offset.replace('+', '')
+        offset = duration_parser.parse(offset)
+        cycle_point = cycle_point + offset
 
-    # remove extraneous whitespace from cycle point
-    value = value.replace(" ", "")
+    return str(cycle_point)
+
+
+def _prev_next(value, now, parser):
+    """Handle prev() and next() syntax.
+
+    Args:
+        value (str):
+            The string containing the prev()/next() stuff.
+        now (metomi.isodatetime.data.TimePoint):
+            A time point to use as the context for resolving the value.
+        parser (metomi.isodatetime.parsers.TimePointParser):
+            A time point parser.
+
+    Returns
+        tuple - (cycle_point, offset)
+
+     """
+    # break down cycle point into constituent parts.
+    direction, tmp = value.split("(")
+    tmp, offset = tmp.split(")")
 
     # correct for year in 'now' if year only,
     # or year and time, specified in input
     if re.search(r"\(-\d{2}[);T]", value):
-        my_now.year = my_now.year + 1
+        now.year = now.year + 1
```

Comment:
I know these lines are from before, but any idea what is meant by "correct for year"? Why is there a need for this correction?

Reply:
No idea whatsoever, perhaps it's rounding up for the next/prev syntax?